hashtable/hashtable.py

```diff
@@ -1,79 +1,86 @@
 class HashTableEntry:
-    """
-    Hash Table entry, as a linked list node.
-    """
-
     def __init__(self, key, value):
         self.key = key
         self.value = value
         self.next = None
 
 
 class HashTable:
-    """
-    A hash table that with `capacity` buckets
-    that accepts string keys
-
-    Implement this.
-    """
-
-    def fnv1(self, key):
-        """
-        FNV-1 64-bit hash function
-
-        Implement this, and/or DJB2.
-        """
+    def __init__(self, capacity):
+        self.capacity = capacity
+        self.storage = [None] * self.capacity
 
+    # def fnv1(self, key):
     def djb2(self, key):
-        """
-        DJB2 32-bit hash function
+        hash = 5381
+        for x in key:
+            hash = (hash * 33) + ord(x)
 
-        Implement this, and/or FNV-1.
-        """
+        return hash & 0xFFFFFFFF
 
     def hash_index(self, key):
-        """
-        Take an arbitrary key and return a valid integer index
-        between within the storage capacity of the hash table.
-        """
-        #return self.fnv1(key) % self.capacity
         return self.djb2(key) % self.capacity
 
     def put(self, key, value):
-        """
-        Store the value with the given key.
 
-        Hash collisions should be handled with Linked List Chaining.
+        index = self.hash_index(key)
+        current_node = self.storage[index]
 
-        Implement this.
-        """
+        if not current_node:
+            self.storage[index] = HashTableEntry(key, value)
+            return
+
+        while current_node:
+            if current_node.key == key:
+                current_node.value = value
+                return
+            elif not current_node.next:
+                current_node.next = HashTableEntry(key, value)
+                return
+            else:
+                current_node = current_node.next
 
     def delete(self, key):
-        """
-        Remove the value stored with the given key.
 
-        Print a warning if the key is not found.
+        index = self.hash_index(key)
+        current_node = self.storage[index]
 
-        Implement this.
-        """
+        if current_node.key == key:
+            self.storage[index] = current_node.next
+            current_node.next = None
+            return
+
+        while current_node.next:
```

Comment:
I believe you could combine everything from line 48 to 60 if you change your while loop condition to `while currrent_node` and checked if the `current_node.key == key`